src/components/EmojiPicker/EmojiPickerMenu/index.js

```diff
@@ -370,6 +385,7 @@ class EmojiPickerMenu extends Component {
 
         // Remove sticky header indices. There are no headers while searching and we don't want to make emojis sticky
         this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: [], highlightedIndex: 0});
+        this.setFirstNonHeaderIndex(newFilteredEmojiList);
```

Comment:
We still need to somehow recalculate `firstNonHeaderIndex` every time emoji list is changed.
We have 2 ways to move this into `highlightAdjacentEmoji` method:
1.
```diff
- const firstNonHeaderIndex = this.state.filteredEmojis.length === this.emojis.length ? this.numColumns : 0;
+ const firstNonHeaderIndex = _.findIndex(this.state.filteredEmojis, item => !item.spacer && !item.header);
```
2. Or ditch tihs function completely and just use 
```diff
- const firstNonHeaderIndex = this.state.filteredEmojis.length === this.emojis.length ? this.numColumns : 0;
+ const firstNonHeaderIndex = this.state.filteredEmojis.length === this.emojis.length ? (this.numColumns * 2) : 0;
```
This also fixes the issue, since `firstNonHeaderIndex` is calculated correctly now, but could in theory break, if we change the number of spacers/headers added to emojis in [addSpacesToEmojiCategories](https://github.com/Expensify/App/blob/8bbc2ecac2c968b54f2b7f954dae940b0590517e/src/libs/EmojiUtils.js#L142), while recalculating it every time doesn't have this disadvantage.

What do you think?

Reply:
Thanks for explaining. Let's do nothing then